elasticsearch/client.go

```diff
@@ -0,0 +1,133 @@
+package elasticsearch
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	dphttp "github.com/ONSdigital/dp-net/http"
+	"github.com/ONSdigital/log.go/log"
+	awsauth "github.com/smartystreets/go-aws-auth"
+)
+
+// ServiceName elasticsearch
+const ServiceName = "elasticsearch"
+
+// Client is an ElasticSearch client containing an HTTP client to contact the elasticsearch API.
+type Client struct {
+	httpCli      dphttp.Clienter
+	url          string
+	serviceName  string
+	signRequests bool
+	indexes      []string
+}
+
+// NewClient returns a new initialised elasticsearch client with the default dp-net/http client
+func NewClient(url string, signRequests bool, maxRetries int, indexes ...string) *Client {
+	httpClient := dphttp.NewClient()
+	httpClient.SetMaxRetries(maxRetries)
+	return NewClientWithHTTPClient(url, signRequests, httpClient, indexes...)
+}
+
+// NewClientWithHTTPClient returns a new initialised elasticsearch client with the provided HTTP client
+func NewClientWithHTTPClient(url string, signRequests bool, httpCli dphttp.Clienter, indexes ...string) *Client {
+	return &Client{
+		httpCli:      httpCli,
+		url:          url,
+		serviceName:  ServiceName,
+		signRequests: signRequests,
+		indexes:      indexes,
+	}
+}
+
+//CreateIndex creates an index in elasticsearch
+func (cli *Client) CreateIndex(ctx context.Context, path string, content []byte) (int, error) {
+
+	url := cli.url + path
+	_, status, err := cli.CallElastic(ctx, url, "PUT", content)
+	if err != nil {
+		return status, err
+	}
+	return status, nil
+}
+
+//DeleteIndex deletes an index in elasticsearch
+func (cli *Client) DeleteIndex(ctx context.Context, path string) (int, error) {
+
+	url := cli.url + path
+	_, status, err := cli.CallElastic(ctx, url, "DELETE", nil)
+	if err != nil {
+		return status, err
+	}
+	return status, nil
+}
+
+//AddDocument adds a document to elasticsearch
+func (cli *Client) AddDocument(ctx context.Context, path string, content []byte) (int, error) {
```

Comment:
See comment above about the path parameter. For the `AddDocument` function, it would need 3 parameters for each segment of the path. I understand this to be `/index/type/id`?

Reply:
I have changed the signature so it is now AddDocument(context context.Context, indexName, indexType, documentID string, document []byte)